protocol/contracts/oud/OudRedeemer.sol

```diff
@@ -0,0 +1,145 @@
+pragma solidity ^0.8.17;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Oud Redeemer (protocol/contracts/oud/OudRedeemer.sol)
+
+import '@openzeppelin/contracts/access/Ownable.sol';
+import '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+import '../interfaces/oud/IOudRedeemer.sol';
+import '../interfaces/oud/Oud__ITempleLineOfCredit.sol';
+import '../interfaces/oud/IOudtoken.sol';
+import '../interfaces/oud/Oud__ITempleERC20Token.sol';
+
+/**
+ * @title Allows for the redemption of Oud + a stable coin for Temple at the Temple Treasury Price index
+ */
+contract OudRedeemer is IOudRedeemer, Ownable {
+  using SafeERC20 for IERC20;
+  error AddressZero();
+  error RedeemAmountZero();
+
+  // @notice Temple token address
+  address public immutable templeToken;
+
+  // @notice Oud token address
+  address public immutable oudToken;
+
+  // @notice Stable token address
+  IERC20 public stable;
+
+  //   @notice The Temple Price Index, used to calculate amount of 'stable' required to mint Temple
```

Comment:
nit: `Treasury Price Index` (not temple, yes easily confused :-))

Why do we have two getters for the same thing? This could be:

```
uint256 public override treasuryPriceIndex;
```

And then you don't need the separate `function treasuryPriceIndex()` below

Reply:
[be6c795](https://github.com/TempleDAO/temple/pull/760/commits/be6c795ffc0b424a146937d3538c914b9a594356)